src/cow-react/modules/limitOrders/pure/ReceiptModal/CurrencyField.tsx

```diff
@@ -0,0 +1,22 @@
+import { Token } from '@uniswap/sdk-core'
+import * as styledEl from './styled'
+import { CurrencySelectButton } from '@cow/modules/swap/pure/CurrencySelectButton'
+
+interface Props {
+  label: string
+  token: Token
+  amount: string | number | undefined
+}
+
+export function CurrencyField({ label, token, amount }: Props) {
+  return (
+    <styledEl.CurrencyField border="rounded-full">
+      <styledEl.FieldTitle>{label}</styledEl.FieldTitle>
+
+      <styledEl.FieldBody>
+        <CurrencySelectButton readonlyMode={true} loading={false} currency={token} />
```

Comment:
P.s. there is `common/pure/CurrencyInputPanel/CurrencyPreview.tsx`, might be helpful

Reply:
Tried to use that one but had some issues adjusting it so I just created a new component.